python_modules/dagster/dagster/core/definitions/config.py

```diff
@@ -35,15 +43,53 @@ class ConfigMapping(
         config_fn (Callable[[dict], dict]): The function that will be called
             to map the composite config to a config appropriate for the child solids.
         config_schema (ConfigSchema): The schema of the composite config.
+        receive_processed_config_values (bool): If true, config values provided to the config_fn
+            will be converted to their dagster types before being passed in. For example, if this
+            value is true, enum config passed to config_fn will be actual enums, while if false,
+            then enum config passed to config_fn will be strings.
     """
 
     def __new__(
         cls,
         config_fn: Callable[[Any], Any],
         config_schema: Any = None,
+        receive_processed_config_values: bool = True,
     ):
         return super(ConfigMapping, cls).__new__(
             cls,
             config_fn=check.callable_param(config_fn, "config_fn"),
             config_schema=convert_user_facing_definition_config_schema(config_schema),
+            receive_processed_config_values=check.bool_param(
+                receive_processed_config_values, "receive_processed_config_values"
+            ),
         )
+
+    def resolve(self, config: Any, config_has_been_validated=False) -> Any:
+        if not config_has_been_validated:
+            if self.receive_processed_config_values:
+                outer_evr = process_config(
+                    self.config_schema.config_type,
+                    config,
+                )
+            else:
+                outer_evr = validate_config(
+                    self.config_schema.config_type,
+                    config,
+                )
+            if not outer_evr.success:
+                raise DagsterInvalidConfigError(
+                    "Error in config mapping ",
+                    outer_evr.errors,
+                    config,
+                )
+
+            outer_config = outer_evr.value
+            if not self.receive_processed_config_values:
+                outer_config = resolve_defaults(
+                    cast(ConfigType, self.config_schema.config_type),
+                    outer_config,
+                ).value
+        else:
+            outer_config = config
+
+        return self.config_fn(outer_config)
```

Comment:
I think just having two different methods would be cleaner than this bool forking most of one fn 

but also this else case seems really bizarre - given the names of the args its unclear why default resolution still doesn't happen